pyomo/dae/set_utils.py

```diff
@@ -0,0 +1,217 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and 
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain 
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+from collections import Counter
+from pyomo.kernel import ComponentSet
+from pyomo.core.base.set import SetProduct
+import pdb
+
+
+def is_explicitly_indexed_by(comp, *sets, **kwargs):
+    """
+    Function for determining whether a pyomo component is indexed by a 
+    set or group of sets.
+
+    Args:
+        comp : some Pyomo component, possibly indexed
+        sets : Pyomo Sets to check indexing by
+
+    Returns:
+        A bool that is True if comp is directly indexed by every set in sets.
+    """
+    if not comp.is_indexed():
+        return False
+    for s in sets:
+        if isinstance(s, SetProduct):
+            msg = ('Checking for explicit indexing by a SetProduct '
+                  'is not supported')
+            raise TypeError(msg)
+
+    expand_all_set_operators = kwargs.pop('expand_all_set_operators', False)
```

Comment:
It is now an optional argument.

Reply:
Oops, I didn't notice that your original implementation was using both `*args` and `**kwargs`. In that case you actually do need to stick with `**kwargs` because I don't think Python 2.7 supports `*args` followed by named keyword arguments without using `**kwargs` (as you'll see in the failing tests). Could you revert this change except for the docstring?

Reply:
Yeah I see that in the failing test, but when I tried locally, it seemed to work with 2.7. I'll revert it.